Move tensor_parallel out to distributed.tensor folder

wanchaol opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89878

This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

wanchaol committed:
Move tensor_parallel out to distributed.tensor folder

This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

fduwjj reviewed (approved):
LGTM, thanks for doing that. Can we ensure that all import of tensor parallel to be moved? Thanks!

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol commented on torch/distributed/tensor/parallel/api.py:
```diff
@@ -10,13 +10,13 @@
     DeviceMesh,
 )
 from torch.distributed._tensor.parallel import TensorParallelMultiheadAttention
```
oops sorry missed that, updating

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol committed:
Update on "Move tensor_parallel out to distributed.tensor folder"


This PR moves tensor parallel from torch.distributed._tensor.parallel
to torch.distributed.tensor.parallel, to prepare for beta release

[ghstack-poisoned]

wanchaol commented:
@pytorchbot merge
